### DISCUSSION TITLE
2.0RC上传图片接口改变了吗？

### DISCUSSION THREAD

**Discussion by Mfweb** [mamoe/mirai#853] [Q&A] [answered]
之前的上传接口报错`NoSuchMethodError`
image = Contact.uploadImage(qqBot.getFriend(sender), getNetFileStream(new URL(imgPath)));
```
Exception in thread "udp" java.lang.NoSuchMethodError: 
net.mamoe.mirai.contact.Contact$Companion.uploadImage$default(Lnet/mamoe/mirai/contact/Contact$Companion;Lnet/mamoe/mirai/contact/Contact;Ljava/io/InputStream;Ljava/lang/String;ILjava/lang/Object;)Lnet/mamoe/mirai/message/data/Image;
```

**Comment by Him188** [comment]
重新编译就好

**Comment by Karlatemp** [comment, marked as answer]
看到 `new` 字眼猜测是 java 语言
```java
ExternalResource.uploadAsImage
```

请完整阅读最新文档 https://github.com/mamoe/mirai/blob/dev/docs/Messages.md#%E6%B6%88%E6%81%AF%E5%85%83%E7%B4%A0

https://github.com/mamoe/mirai/blob/7d3bcbeb1ca1f664052176ab04bd1c4d35ae57f8/mirai-core-api/src/commonMain/kotlin/message/data/Image.kt#L58-L64

**Comment by Him188** [comment]
这个接口其实还在

https://github.com/mamoe/mirai/blob/7d3bcbeb1ca1f664052176ab04bd1c4d35ae57f8/mirai-core-api/src/commonMain/kotlin/contact/Contact.kt#L91-L130

使用 `Contact.sendImage(contact, inputStream)`

**Reply by Him188** [reply]
发现是 KJBB 没有处理 `@JvmOverloads` (https://github.com/Him188/kotlin-jvm-blocking-bridge/issues/6)

**Reply by Him188** [reply]
已经修复, 稍后会发布一个修复版本

**Reply by Him188** [reply]
可使用 `2.0.0-dev-1`
